pkg/merger/merger.go

```diff
@@ -206,3 +211,17 @@ func MergeAndUpdate(ctx context.Context, client mergeClient, mets *Metrics, list
 	finish.Success()
 	return result, nil
 }
+
+func recordLastModified(attrs *storage.ReaderObjectAttrs, mets *Metrics, source string) {
+	if attrs != nil {
+		lastModified := attrs.LastModified
+		diff := time.Since(lastModified)
+		if mets != nil {
+			mets.LastModified.Set(int64(diff.Seconds()), source)
+			logrus.WithFields(logrus.Fields{
```

Comment:
I would log this information even when metrics are disabled. Info is a good log level.

Reply:
Consider moving this `logrus.WithFields...` call outside the if statement instead. Writing the shards and their last-update-time to logs is still useful, even if dedicated metrics are disabled or missing.

"Mets is nil" is not as useful.